src/utils/decorators/tooltip-decorator/tooltip-decorator.js

```diff
@@ -163,198 +122,145 @@ const TooltipDecorator = (ComposedComponent) => {
       clearTimeout(this._hideTooltipTimeout);
 
+      if (this.state.isVisible) return;
+
       this._showTooltipTimeout = setTimeout(() => {
         this.setState({ isVisible: true });
         this.positionTooltip();
       }, 100);
     };
 
-    /**
-     * Hides tooltip
-     *
-     * @method onHide
-     * @return {void}
-     */
     onHide = () => {
       clearTimeout(this._showTooltipTimeout);
+      clearTimeout(this._hideTooltipTimeout);
+
+      if (!this.state.isVisible) return;
 
       this._hideTooltipTimeout = setTimeout(() => {
         this.setState({ isVisible: false });
       }, 100);
     };
 
-    /**
-     * Returns the DOM node of the target.
-     *
-     * @method getTarget
-     * @return {DOM node}
-     */
     getTarget = () => {
       return ReactDOM.findDOMNode(this._target); // eslint-disable-line react/no-find-dom-node
     }
 
-    /**
-     * Returns the DOM node of the tooltip.
-     *
-     * @method getTooltip
-     * @return {DOM node}
-     */
     getTooltip = () => {
       return ReactDOM.findDOMNode(this._tooltip); // eslint-disable-line react/no-find-dom-node
     }
 
-    /**
-     * Calculates position for tooltip, first result cached.
-     *
-     * @method calculatePosition
-     * @param {Object} tooltip
-     * @param {Object} target
-     * @return {Object} shifts calculated
-     */
     calculatePosition = (tooltip, target) => {
       if (this._memoizedShifts) { return this._memoizedShifts; }
 
       const tooltipWidth = tooltip.offsetWidth,
           tooltipHeight = tooltip.offsetHeight,
-          pointerDimension = 15,
-          // hardcode height & width since span has no dimensions
-          pointerOffset = 11,
           targetWidth = target.offsetWidth,
           targetHeight = target.offsetHeight,
           targetRect = target.getBoundingClientRect(),
           offsetY = window.pageYOffset,
           targetTop = targetRect.top + offsetY,
           targetBottom = targetRect.bottom + offsetY,
           targetLeft = targetRect.left,
-          targetRight = targetRect.right;
-
-      const shifts = {
-        verticalY: targetTop - tooltipHeight - (pointerDimension * 0.5),
-        verticalBottomY: targetBottom + (pointerDimension * 0.5),
-        verticalCenter: (targetLeft - (tooltipWidth * 0.5)) + (targetWidth * 0.5),
-        verticalRight: (targetLeft + pointerDimension + pointerOffset) - tooltipWidth,
-        verticalLeft: targetLeft - (pointerDimension * 0.5),
-        rightHorizontal: targetRight + (0.5 * pointerDimension),
-        leftHorizontal: targetLeft - (pointerDimension * 0.5) - tooltipWidth,
-        sideTop: targetTop - pointerOffset,
-        sideBottom: (targetTop - tooltipHeight) + targetHeight + pointerOffset,
-        sideCenter: (targetTop + (targetHeight * 0.5)) - (tooltipHeight * 0.5)
+          targetRight = targetRect.right,
+          targetXCenter = targetWidth / 2,
+          targetYCenter = targetHeight / 2;
```

Comment:
The names `targetXCenter` and `targetYCenter` sound like coordinates, but they're not. Maybe renaming them to `targetHalfWidth` and `targetHalfHeight` would make their purpose clearer?

Reply:
good point, changed

Reply:
Thanks :white_check_mark: